docs/guide/content_rendering.md

```diff
@@ -1073,31 +1190,31 @@ In this example, as the `ViewController` is not applied, the custom controller t
 
 namespace Acme\TestBundle\Controller;
 
-use Symfony\Component\HttpFoundation\Response;
 use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\Core\MVC\Symfony\View\ContentView;
 
 class DefaultController extends Controller
 {
-    public function viewFolderAction( $locationId, $layout = false, $params = array() )
+    public function viewFolderAction(ContentView $view)
     {
-        $repository = $this->getRepository();
-        $location = $repository->getLocationService()->loadLocation( $locationId );
-        // Check if Content is not already passed. Can be the case when using content preview.
-        $content = isset( $params['content'] ) ? $params['content'] : $repository->getContentService()->loadContentByContentInfo( $location->getContentInfo() )
-        $response = new Response();
-        $response->headers->set( 'X-Location-Id', $locationId );
-        // Set caching for 1h and make the cache vary on user hash
-        $response->setSharedMaxAge( 3600 );
-        $response->setVary( 'X-User-Hash' );
-        return $this->render(
+        $location = $view->getLocation();
+        $content = $view->getContent();
+    
+        $response = $this->render(
             'AcmeTestBundle::custom_controller_folder.html.twig',
-            array(
+            [
                 'location' => $location,
                 'content' => $content,
                 'foo' => 'Hey world!!!',
-                'osTypes' => array( 'osx', 'linux', 'windows' )
-            ) + $params
+                'osTypes' => ['osx', 'linux', 'windows']
+            ]
         );
+    
+        // Set caching for 1h and make the cache vary on user hash
+        $response->setSharedMaxAge(3600);
```

Comment:
What do we want to show here?

Casue by default CotnentView will be cached, by user hash, unless you opt out.

So we can show:
- addParameters(): To add parameters like the one above
- setResponse(): For setting custom repsponse headers
- setCacheEnabled(false): For disabling caching (but of course not recommended)

So example here could be written in a way that it still returns view, depending on what we want to show here.

Reply:
I changed this example to setting custom response header instead of dealing with cache. I wanted to show how to modify the `Response` object, so it was a good and safe choice.